File: frontends/main/src/app/configurator/services/api/event-model-rights.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {EntityRight} from '@core/model/entity-right';

@Injectable({providedIn: 'root'})
export class EventModelRightsService {
	constructor(private http: HttpClient) {}

	getRights(projectId: string): Observable<Record<string, Record<string, EntityRight>>> {
		return this.http.get<Record<string, Record<string, EntityRight>>>(
			`/api/superuser/configurator/projects/${projectId}/config/event-model-rights`
		);
	}

	saveRights(projectId: string, rights: Record<string, Record<string, EntityRight>>): Observable<void> {
		return this.http.put<void>(
			`/api/superuser/configurator/projects/${projectId}/config/event-model-rights`,
			rights
		);
	}
}

File: backend/src/main/java/ch/rodano/core/services/bll/configurator/EventModelRightsService.java
package ch.rodano.core.services.bll.configurator;

import java.util.Map;
import java.util.UUID;

import ch.rodano.api.config.EntityRightDTO;

public interface EventModelRightsService {

	Map<UUID, Map<UUID, EntityRightDTO>> getEventModelRights(UUID projectId);

	void saveEventModelRights(UUID projectId, Map<UUID, Map<UUID, EntityRightDTO>> rights);
}
